### PR TITLE
Separate out calibration script into modules

### LINKED ISSUE #62: Separate out the calibration script into separate modules
The vesuvio calibration script file is large, and is only getting longer with refactoring and the adding of functionality.

In order to make it easier to work with, this script should be split out into separate modules to hole the `EVSCalibrationFit` and `EVSCalibrationAnalysis` classes.

Changes will need to be made to the system/unit tests in order to allow for this.

### PR DESCRIPTION
**Pull request by MialLewis**
**Description of work:**

This PR separates out the calibration script [calibrate_vesuvio_uranium_martyn_MK5.py](https://github.com/mantidproject/vesuvio/pull/61/files#diff-cd45feda55bf17d046f59e13c4c3e96339c643a4d6fd0759ff97f6cd4bdf4f97) into separate modules.

This script file was overly large and was getting hard to work with.

Whilst doing this, legacy formatting issues have been resolved (indentation is now 4 as per pep8 etc.).

Old versions of the script and system tests have been removed.

Given the separated scripts are now "harder" to use, a `README` has been added.

**To test:**

Using the new README as a guide:
1. Ensure system tests pass.
2. Ensure unit tests pass
3. Load the algorithms into `mantid` using the new script.

Resolves #62

### CODE REVIEW

File: unpackaged/vesuvio_calibration/load_calibration_algorithms.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
+"""
+  This script imports and subscribes calibration algorithms for the VESUVIO spectrometer.
+
+  Two Mantid algorithms are provided: EVSCalibrationFit and EVSCalibrationAnalysis.
+
+  EVSCalibrationFit is used to fit m peaks to n spectra. The positions of the peaks are esitmated using the supplied instrument parameter
+  file and the d-spacings of the sample (if provided). Support is provided for both Voigt and Gaussian functions.
+
+  EVSCalibrationAnalysis uses EVSCalibrationFit to calculate instrument parameters using the output of the fitting and the and an existing
```
Comment (MohamedAlmaki) [inline unpackaged/vesuvio_calibration/load_calibration_algorithms.py:9]: There is a typo in "and the and an" and the sentence is a little bit confusing

File: unpackaged/vesuvio_calibration/tests/system/test_system_analysis.py [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+    def _assert_parameters_match_expected(self, params_table, tolerances=None):
+        rel_tol_theta, rel_tol_L1 = self._extract_tolerances(tolerances)
+        theta_errors = self._assert_theta_parameters_match_expected(params_table, rel_tol_theta)
+        L1_errors = self._assert_L1_parameters_match_expected(params_table, rel_tol_L1)
+
+        if theta_errors or L1_errors:
+            raise AssertionError(f"Theta: {theta_errors})\n L1: {L1_errors}")
+
+    @staticmethod
+    def _extract_tolerances(tolerances: dict) -> (dict, dict):
```
Comment (MohamedAlmaki) [inline unpackaged/vesuvio_calibration/tests/system/test_system_analysis.py:237]: Should we remove the type annotation to maintain consistency with the other functions in the file? what do you think?

File: unpackaged/vesuvio_calibration/tests/system/test_system_fit.py [hunk 3]
```diff
@@ -0,0 +1,226 @@
+import unittest
+import numpy as np
+import scipy.constants
+import scipy.stats
+
+from mantid.api import WorkspaceGroup, AlgorithmFactory
+from mantid.simpleapi import LoadVesuvio, LoadRaw, mtd, ConvertToDistribution
```
Comment (MohamedAlmaki) [inline unpackaged/vesuvio_calibration/tests/system/test_system_fit.py:7]: Same here, LoadVesuvio, LoadRaw, ConvertToDistribution are unused

**Review by MohamedAlmaki — CHANGES_REQUESTED** [review]
The tests have passed successfully. I've added a few comments on the code. But there is a problem with running the script on Mantid. I get this error when I try to run the script:

File: unpackaged/vesuvio_calibration/calibration_scripts/calibrate_vesuvio_analysis.py [hunk 4]
```diff
@@ -0,0 +1,510 @@
...
+
+          @param table_name - name of table containing fitted parameters for the peak centres
+          @param spec_list - spectrum range to calculate t0 for.
+        """
+
+        t0 = EVSMiscFunctions.read_table_column(self._current_workspace, 't0', spec_list)
+        L0 = EVSMiscFunctions.read_table_column(self._current_workspace, 'L0', spec_list)
+        L1 = EVSMiscFunctions.read_table_column(self._param_table, 'L1', spec_list)
+        L1_nan_to_num = np.nan_to_num(L1)
+        spec = EVSMiscFunctions.read_table_column(self._current_workspace, 'Spectrum', spec_list)
```
[the remark this answers is not in the document]
Comment (MialLewis) [reply unpackaged/vesuvio_calibration/calibration_scripts/calibrate_vesuvio_analysis.py:283]: Left over from scientist who wrote the algorithm, I've removed it.

File: unpackaged/vesuvio_calibration/calibrate_vesuvio_uranium_martyn_MK5.py [hunk 5]
```diff
@@ -1,1845 +0,0 @@
...
-    else:
-        raise ValueError("Unsupported fit function type: %s" % function_type)
-
-    return {k: v+error_str for k,v in func_header.items()}
-
-#----------------------------------------------------------------------------------------
-
-
-class EVSCalibrationFit(PythonAlgorithm):
-
```
[the remark this answers is not in the document]
Comment (MialLewis) [reply unpackaged/vesuvio_calibration/calibrate_vesuvio_uranium_martyn_MK5.py:241]: After this PR they won't be. Previously the scientist was keeping multiple versions of the script, but these have been deleted and consolidated into `calibrate_vesuvio_analysis.py` etc.

File: unpackaged/vesuvio_calibration/calibration_scripts/calibrate_vesuvio_analysis.py [hunk 6]
```diff
@@ -0,0 +1,510 @@
...
+     FloatArrayProperty
+from mantid.api import FileProperty, FileAction, ITableWorkspaceProperty, PropertyMode, Progress, TextAxis, PythonAlgorithm,\
+     AlgorithmManager
+from mantid.simpleapi import CreateEmptyTableWorkspace, DeleteWorkspace, CropWorkspace, RebinToWorkspace, Divide,\
+     ReplaceSpecialValues, FindPeaks, GroupWorkspaces, mtd, Plus, LoadVesuvio, LoadRaw, ConvertToDistribution,\
+     FindPeakBackground, ExtractSingleSpectrum, SumSpectra, AppendSpectra, ConvertTableToMatrixWorkspace,\
+     ConjoinWorkspaces, Transpose, PlotPeakByLogValue, CloneWorkspace, MaskDetectors,\
+     ExtractUnmaskedSpectra, CreateWorkspace, RenameWorkspace
+from unpackaged.vesuvio_calibration.calibration_scripts.calibrate_vesuvio_helper_functions import EVSGlobals, EVSMiscFunctions
+
```
[the remark this answers is not in the document]
Comment (MialLewis) [reply unpackaged/vesuvio_calibration/calibration_scripts/calibrate_vesuvio_analysis.py:12]: Removed.

File: unpackaged/vesuvio_calibration/calibration_scripts/calibrate_vesuvio_fit.py [hunk 7]
```diff
@@ -0,0 +1,1080 @@
...
+from mantid.api import FileProperty, FileAction, ITableWorkspaceProperty, PropertyMode, Progress, TextAxis, PythonAlgorithm,\
+     WorkspaceFactory, AnalysisDataService
+from mantid.simpleapi import CreateEmptyTableWorkspace, DeleteWorkspace, CropWorkspace, RebinToWorkspace, Divide,\
+     ReplaceSpecialValues, FindPeaks, GroupWorkspaces, mtd, Plus, LoadVesuvio, LoadRaw, ConvertToDistribution,\
+     FindPeakBackground, ExtractSingleSpectrum, SumSpectra, AppendSpectra, ConvertTableToMatrixWorkspace,\
+     ConjoinWorkspaces, Transpose, PlotPeakByLogValue, CloneWorkspace, Fit, MaskDetectors,\
+     ExtractUnmaskedSpectra, CreateWorkspace
+from functools import partial
+from unpackaged.vesuvio_calibration.calibration_scripts.calibrate_vesuvio_helper_functions import EVSGlobals, EVSMiscFunctions
+
```
[the remark this answers is not in the document]
Comment (MialLewis) [reply unpackaged/vesuvio_calibration/calibration_scripts/calibrate_vesuvio_fit.py:13]: Removed.

File: unpackaged/vesuvio_calibration/load_calibration_algorithms.py (hunk 1 above)
Comment (MialLewis) [reply unpackaged/vesuvio_calibration/load_calibration_algorithms.py:9]: Ammended.

File: unpackaged/vesuvio_calibration/tests/system/test_system_fit.py (hunk 3 above)
Comment (MialLewis) [reply unpackaged/vesuvio_calibration/tests/system/test_system_fit.py:7]: Removed.

**Comment by MialLewis** [comment]
> The tests have passed successfully. I've added a few comments on the code. But there is a problem with running the script on Mantid. I get this error when I try to run the script: image

For some reason this was working on linux but not windows. It should now work on windows, though you need to run the tests from a different directory (see the README).

**Review by MohamedAlmaki — APPROVED** [review]
LGTM, the script is now working well and the tests have passed successfully.

**Pull request merged**